### PR TITLE
857/chart style

### PR DESCRIPTION
**Pull request by hanbyul-here**
**Related Ticket:** #857 

### Description of Changes
Change new E&A Chart style

### Notes & Questions About Changes
Part of #857
The list of changes specified in Figma: https://www.figma.com/file/9INQauBWhiRxvOWDGhRrxO/US-GHG-Center?type=design&node-id=1127-2201&mode=design&t=VNfErIVXtEgHmbZi-0
- [x] On the chart, average and standard deviation are shown by default.
- [x] Standard deviation is shown as a grey band around the average line. 
- [x] Median, min and max are optional, accessed through the “Chart options” icon. (*Unless editor specified which analysis to show)
- [x] Min and max are shown as grey dotted lines.
- [x] The y-axis is shown on the gray area outside the analysis period.
- [x] The y-axis should always show the zero line
- [x] Always expand with yaxis. Get rid of 'collapse' button
- [x] Optional:  The chart actions icons are:”Download chart”, “Chart options” and “Collapse”. Collapse is a local action and collapses the current chart only. - **only implemented "Chart Options" for now**

Question: What should we do for circles that mark the time unit? If I remember correctly, we added it because of the requests from stakeholders.

### Validation / Testing

To check the changes, run analysis [on this page](https://deploy-preview-903--veda-ui.netlify.app/exploration?datasets=%5B%7B%22id%22%3A%22no2-monthly%22%2C%22settings%22%3A%7B%22isVisible%22%3Atrue%2C%22opacity%22%3A100%2C%22analysisMetrics%22%3A%5B%7B%22id%22%3A%22mean%22%2C%22label%22%3A%22Average%22%2C%22chartLabel%22%3A%22Avg%22%2C%22themeColor%22%3A%22infographicB%22%7D%2C%7B%22id%22%3A%22std%22%2C%22label%22%3A%22St+Deviation%22%2C%22chartLabel%22%3A%22STD%22%2C%22themeColor%22%3A%22infographicD%22%7D%5D%7D%7D%2C%7B%22id%22%3A%22no2-monthly-diff%22%2C%22settings%22%3A%7B%22isVisible%22%3Atrue%2C%22opacity%22%3A100%2C%22analysisMetrics%22%3A%5B%7B%22id%22%3A%22mean%22%2C%22label%22%3A%22Average%22%2C%22chartLabel%22%3A%22Avg%22%2C%22themeColor%22%3A%22infographicB%22%7D%2C%7B%22id%22%3A%22std%22%2C%22label%22%3A%22St+Deviation%22%2C%22chartLabel%22%3A%22STD%22%2C%22themeColor%22%3A%22infographicD%22%7D%5D%7D%7D%5D&date=2023-08-31T15%3A00%3A00.000Z&aois=%5B%22le%7B_Tga%7EbDayfqWpxoNt%60%7EvEl_rpDpuaoJjb%7CsB%22%2C%228cd277%22%2Cfalse%5D&dateRange=2022-08-09T15%3A00%3A00.000Z%7C2023-09-30T15%3A00%3A00.000Z)

### CODE REVIEW

**Review by faustoperez — CHANGES_REQUESTED** [review]
Great work Hanbyul!

One thing about the tooltip behavior: if someone hovers on the chart there should always be an "active node" so that the tooltip is present, no matter the mouse position. At the moment, it disappears if the mouse position is not near a node. I think that's the reason the little dots on the line charts were included in the first place. 

Frame 2516



Proposed fix: the tooltip should always be present when hovering over the chart

File: app/scripts/components/analysis/define/index.tsx [hunk 1]
```diff
@@ -431,7 +431,7 @@ export default function Analysis() {
                     value={end ? dateToInputFormat(end) : ''}
                     onChange={onEndDateChange}
                     min={dateToInputFormat(start)}
-                    max='2022-12-31'
+                    max='2023-12-31'
```
Comment (sandrahoang686) [inline app/scripts/components/analysis/define/index.tsx:434]: Should the year be dynamic with current - 1 ?

**Comment by sandrahoang686** [comment]
Did a quick lookover, code looks fine to me! But I can take a more detailed look later

File: app/scripts/components/analysis/define/index.tsx (hunk 1 above)
Comment (hanbyul-here) [reply app/scripts/components/analysis/define/index.tsx]: This is a good catch. This value seemed to be hard-coded with an intention https://github.com/NASA-IMPACT/veda-ui/commit/a40244d3276809b665f535209ea67fcdd39785a4#diff-7e1c0b42a5124126043b8ee857dd483459ebc27bd003244d0793649c817c1bbcR421 but I am not sure why, I would wait until @j08lue can confirm if we can chang the date. I revereted this change.

**Comment by hanbyul-here** [comment]
@faustoperez I handled the hovering issue 💦

**Comment by faustoperez** [comment]
@hanbyul-here Great job, I wasn't able to replicate the missing tooltip behavior ✊ 

One thing that's missing is the vertical line on the active node so users have a reference of the hover point, as in the designs. Also, the dots for the St Deviation are missing.

Implementation:
image

Designs:
image

Thank you! 💐

**Comment by hanbyul-here** [comment]
@sandrahoang686 This PR is ready for reviews 🙇

**Comment by j08lue** [comment]
Looks amazing! 😍 This is a huge improvement to the scientific value of these charts.

Please note - labels on the chart value tooltip should be spelled out, please:

1. "Average"
2. "St Deviation"

image

image

File: app/scripts/components/exploration/components/datasets/dataset-chart.tsx [hunk 2]
```diff
@@ -123,19 +161,93 @@ export function DatasetChart(props: DatasetChartProps) {
...
   prop: string;
   data: any[];
   color: string;
   isVisible: boolean;
-  isExpanded: boolean;
   highlightDate?: Date;
 }
+interface DateLineProps extends DataAreaProps {
+  style?: any;
+}
```
Comment (sandrahoang686) [inline app/scripts/components/exploration/components/datasets/dataset-chart.tsx:176]: Did you mean to keep this any type?

File: app/scripts/components/exploration/constants.ts [hunk 3]
```diff
@@ -11,3 +11,10 @@ export const emptyDateRange = {
 };
 
 export const MAX_QUERY_NUM = 300;
+
+// @TECH-DEBT As we do not have a new design system that can accommodate the needs for colors
+// We put colors here for now
+export const AXIS_BG_COLOR = '#F0F0F0';
+export const HIGHLIGHT_LINE_COLOR = '#CCCCCC';
+export const FADED_TEXT_COLOR = '#83868A';
+export const TEXT_TITME_BG_COLOR = '#FAFAFA';
```
Comment (sandrahoang686) [inline app/scripts/components/exploration/constants.ts:20]: type

File: mock/datasets/no2.data.mdx [hunk 4]
```diff
@@ -99,11 +99,11 @@ layers:
...
-      metrics:
-        - min
-        - max
-        # dummy value to make sure non existent values are sagfely discarded
-        - non-existent
+      # metrics:
+      #   - min
+      #   - max
+      #   # dummy value to make sure non existent values are sagfely discarded
+      #   - non-existent
```
Comment (sandrahoang686) [inline mock/datasets/no2.data.mdx:106]: remove?

Comment (hanbyul-here) [reply mock/datasets/no2.data.mdx]: I commented out  this part for testing; I reverted this change.

**Review by faustoperez — APPROVED** [review]
Looking good Hanbyul great work 🙌

**Comment by j08lue** [comment]
This also closed
- ttps://github.com/NASA-IMPACT/veda-ui/issues/857

**Pull request merged**